rasa/utils/tensorflow/layers.py

```diff
@@ -121,6 +126,51 @@ def __init__(self, reg_lambda: float = 0, **kwargs: Any) -> None:
 
         super().__init__(kernel_regularizer=regularizer, **kwargs)
 
+    def get_units(self) -> int:
+        """Returns number of output units."""
+        return self.units
+
+    def get_kernel(self) -> tf.Tensor:
+        """Returns kernel tensor."""
+        return self.kernel
+
+    def get_bias(self) -> Union[tf.Tensor, None]:
+        """Returns bias tensor."""
+        if self.use_bias:
+            return self.bias
+        return None
+
+    def get_feature_type(self) -> Union[Text, None]:
+        """Returns a feature type of the data that's fed to the layer.
+
+        In order to correctly return a feature type, the function heavily relies
+        on the name of `DenseForSparse` layer being in the following format:
+        f"sparse_to_dense.{attribute}_{feature_type}".
+
+        Returns:
+            feature type of dense layer.
+        """
+        for feature_type in [FEATURE_TYPE_SENTENCE, FEATURE_TYPE_SEQUENCE]:
+            if feature_type in self.name:
+                return feature_type
+        return None
+
+    def get_attribute(self) -> Union[Text, None]:
+        """Returns an attribute of the data that's fed to the layer.
+
+        In order to correctly return an attribute, the function heavily relies
+        on the name of `DenseForSparse` layer being in the following format:
+        f"sparse_to_dense.{attribute}_{feature_type}".
+
+        Returns:
+            attribute of dense layers.
+        """
+        attribute = None
+        metadata = self.name.split(".")
+        if len(metadata) > 1:
+            attribute = metadata[1].split("_")[0]
```

Comment:
Couldn't answer your question on the original thread for some reason. Answering it here:

Because `DenseForSparse` layer can be used by codebases outside of Rasa as well. We can only take guarantee for specific behaviour:
1. Name being in a certain format
2. Attribute being one of the elements of the list.

If the above conditions are not met, codebase cannot find the attribute.